code/clue.js

```diff
@@ -41,19 +198,59 @@ const randomSelector = array => {
 // With a killer, a weapon and a room.
 // The values will be set later.
 
+const mystery = {
+  killer: null,
+  weapons: null,
+  rooms: null
+}
+
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+
+// This will randomly select a killer from the suspects. And add that to the mystery object.
+mystery.killer = randomSelector(suspects)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage'). src = mystery.killer.image
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age} years old`
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`
 }
 
+
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+const pickWeapons = () => {
+
+  mystery.weapons = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapons.color
+  document.getElementById('weaponImage'). src = mystery.weapons.image
+  document.getElementById('weaponName').innerHTML = `${mystery.weapons.name}`
+}
+
+
+const pickRoom = () => {
+
+  mystery.room = randomSelector(rooms)
+  document.getElementById('roomName').innerHTML = `${mystery.room}`
+}
+
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+document.getElementById('killerCard').onclick = pickKiller
+document.getElementById('weaponCard').onclick = pickWeapons
+document.getElementById('roomCard').onclick = pickRoom
```

Comment:
Nice! I really like that you grouped the event handlers together.  I feel like I can trust that I've seen all of the `onclick` which I need to be aware of on this page